### PR TITLE
BPLAT-9528 | Add Vualto Widewine support for Android

### PR DESCRIPTION
**Pull request by cywong-amino**
https://jira.aminocom.com/browse/BPLAT-9528

### CODE REVIEW

**Comment by chloeching-amino** [comment]
reviewing...

File: android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java [hunk 1]
```diff
@@ -353,7 +370,9 @@ public void run() {
 
                     DefaultAllocator allocator = new DefaultAllocator(true, C.DEFAULT_BUFFER_SEGMENT_SIZE);
                     DefaultLoadControl defaultLoadControl = new DefaultLoadControl(allocator, minBufferMs, maxBufferMs, bufferForPlaybackMs, bufferForPlaybackAfterRebufferMs, -1, true);
-                    player = ExoPlayerFactory.newSimpleInstance(getContext(), trackSelector, defaultLoadControl);
+                    DrmSessionManager<FrameworkMediaCrypto> drmSessionManager = drmSessionManager = buildDrmSessionManager(self.drmUUID, self.srcUri, self.drmLicenseUrl, self.drmToken);
```
Comment (chloeching-amino) [inline android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java]: Duplicated `drmSessionManager =`?
```suggestion
                    DrmSessionManager<FrameworkMediaCrypto> drmSessionManager = buildDrmSessionManager(self.drmUUID, self.srcUri, self.drmLicenseUrl, self.drmToken);
```

File: android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java [hunk 2]
```diff
@@ -398,6 +417,51 @@ public void run() {
...
+        try {
+            assetConfiguration = new AssetConfiguration.Builder()
+                    .tokenWith(drmToken)
+                    .licenceUrlWith(licenseServerUrl)
+                    .kidProviderWith(
+                            new HttpKidSource(
+                                    new URL(streamUrl.toString())
+                            )
+                    )
+                    .build();
```
Comment (chloeching-amino) [inline android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java:444]: Indentation :thinking:

File: android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java [hunk 3]
```diff
@@ -398,6 +417,51 @@ public void run() {
...
+            return null;
+        }
+        WidevineCallback callback = new WidevineCallback(assetConfiguration);
+        try {
+            return new DefaultDrmSessionManager<>(drmUUID,
+                    FrameworkMediaDrm.newInstance(drmUUID),
+                    callback,
+                    null,
+                    new Handler(),
+                    this);
```
Comment (chloeching-amino) [inline android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java:457]: Indentation :thinking:

File: android-exoplayer/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -11,4 +11,12 @@
 
   <string name="unrecognized_media_format">Unrecognized media format</string>
 
+  <string name="error_drm_not_supported">Protected content not supported on API levels below 18</string>
+
+  <string name="error_drm_unknown">An unknown DRM error occurred</string>
+
+  <string name="error_drm_unsupported_scheme">This device does not support the required DRM scheme</string>
+
+  <string name="error_drm_session">An error with DRM session occurred</string>
```
Comment (chloeching-amino) [inline android-exoplayer/src/main/res/values/strings.xml]: Would it be better to add the error keys in the messages? Then RN side can grep the error more easily, and no need to worry about the future changes of the error message sentences. :thinking:

File: android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java (hunk 2 above)
Comment (cywong-amino) [reply android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java:444]: I am just sticking to the automatic format.

**Pull request merged**